.github/workflows/build_and_deploy.yml

```diff
@@ -109,13 +111,55 @@ jobs:
           # Downloads to '/home/runner/work/docs/docs/html'
           path: html
 
-      - name: Deploy Docs
+      - name: Deploy Dev Docs
         if: (github.event_name == 'push' || github.event_name == 'workflow_dispatch') && (startsWith(github.ref, 'refs/tags/v') || startsWith(github.ref, 'refs/heads/main'))
         uses: peaceiris/actions-gh-pages@v3
         with:
+          # Note that GITHUB_TOKEN has no permission to access to external repositories.
+          # When you use deploy_key, set your private key to the repository which
+          # includes this action as a secret, and set your public key to your external repository.
           deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
           external_repository: napari/napari.github.io
           publish_dir: ./html
           publish_branch: gh-pages
           destination_dir: ${{ github.event.inputs.target_directory || 'dev' }}
           cname: napari.org
+          keep_files: true
+
+      # Because we are using two deploy actions, we need to reset the ssh-agent
+      # to avoid the following error:
+      # unix_listener: cannot bind to path /tmp/ssh-auth.sock: Address already in use
+      # See https://github.com/peaceiris/actions-gh-pages/issues/909
+      - name: Reset ssh agent
+        if: (github.event_name == 'release' || (github.event_name == 'workflow_dispatch' && github.event.inputs.target_directory != 'dev'))
+        run: killall ssh-agent
+
+      # Because the actions-gh-pages triggers a pages deployment, it can be some
+      # time before the repo is updated. We need to wait for the deployment to
+      # complete before we can create a second deployment, or we will have a git
+      # conflict. The 5m sleep is completely arbitrary.
+      - name: Wait for first deployment
+        if: (github.event_name == 'release' || (github.event_name == 'workflow_dispatch' && github.event.inputs.target_directory != 'dev'))
+        run: sleep 5m
+
+      - name: Deploy Release Docs
+        if: (github.event_name == 'release' || (github.event_name == 'workflow_dispatch' && github.event.inputs.target_directory != 'dev'))
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
+          external_repository: napari/napari.github.io
+          publish_dir: ./html
+          publish_branch: gh-pages
+          destination_dir: ${{ github.event.inputs.target_directory || github.event.release.tag_name }}
```

Comment:
This could potentially create a wrong folder if the wrong version string is passed as input to the action.

Reply:
I'm not sure how to validate this entry :(